tailscale/client.go

```diff
@@ -333,6 +332,62 @@ func (c *Client) DNSNameservers(ctx context.Context) ([]string, error) {
 	return resp["dns"], nil
 }
 
+// SplitDnsRequest is a map from domain names to a list of nameservers.
+type SplitDnsRequest map[string][]string
+
+// SplitDnsResponse is a map from domain names to a list of nameservers.
+type SplitDnsResponse SplitDnsRequest
+
+// UpdateDNSSplitDNS updates the split DNS settings for a tailnet using the
+// passed in SplitDnsRequest object. This is a PATCH operation that performs
+// partial updates of the underlying data structure.
+//
+// Mapping a domain to a nil slice in the request will unset the nameservers
+// associated with that domain.
+func (c *Client) UpdateDNSSplitDNS(ctx context.Context, request SplitDnsRequest) error {
+	const uriFmt = "/api/v2/tailnet/%v/dns/split-dns"
+
+	req, err := c.buildRequest(ctx, http.MethodPatch, fmt.Sprintf(uriFmt, c.tailnet), requestBody(request))
+	if err != nil {
+		return err
+	}
+
+	return c.performRequest(req, nil)
```

Comment:
Similarly following existing convention where we do not seem to return the response object from non-GET endpoints. Can add a return of the actual object here instead if we think that makes sense.

Reply:
I think maybeee we should return it? Since this is a patch, it's extra useful to know what the new updated config is. But no strong preference

Reply:
Ahh good call! Agree that it makes sense for the PATCH endpoint specifically, will adjust.